flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -171,23 +181,25 @@ public void deregisterApplication(
                 parameters.getTaskManagerCPU());
 
         final CompletableFuture<Void> createPodFuture =
-                flinkKubeClient.createTaskManagerPod(taskManagerPod);
+                flinkKubeClient.createTaskManagerPod(taskManagerSpec);
 
+        final KubernetesPod finalTaskManagerPod = taskManagerPod;
+        final String finalPodName = podName;
```

Comment:
Just tried it. Because we can only get `taskManagerPod`, `podName` in the try-catch block. If we put only `buildTaskManagerKubernetesPod` in there, we need to move `taskManagerSpec`, `podName`, `taskManagerPod` out of the block, i.e.,

```java
KubernetesTaskManagerSpecification taskManagerSpec = null;
KubernetesPod taskManagerPod = null;
String podName = null;
try {
  taskManagerSpec = ...
  ...
} catch (Exception e) {
  requestResourceFuture.completeExceptionally(e);
}
```

Due to "Variable used in lambda expression should be final or effectively final", I need to have copies of them.